WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -797,6 +811,65 @@ extension ProductSelectorViewModel {
     }
 }
 
+// MARK: - Tracking
+private extension ProductSelectorViewModel {
+    func trackConfirmButtonTapped(with productsCount: Int) {
+        let trackingSources = Array(productIDTrackingSources.values.map { $0.rawValue })
+        let areFiltersActive = filtersSubject.value.numberOfActiveFilters > 0
+        analytics.track(event: WooAnalyticsEvent.Orders.orderCreationProductSelectorConfirmButtonTapped(productCount: productsCount,
+                                                                                                        sources: trackingSources,
+                                                                                                        isFilterActive: areFiltersActive))
+    }
+
+    func updateTrackingSourceAfterSelectionStateChangedForProduct(with productID: Int64) {
+        guard productIDTrackingSources[productID] == nil else {
+            productIDTrackingSources.removeValue(forKey: productID)
+
+            return
+        }
+
+        if let trackingSource = retrieveTrackingSource(for: productID) {
+            productIDTrackingSources[productID] = trackingSource
+        }
+    }
+
+    func updateTrackingSourceAfterSelectionStateChangedForProduct(with productID: Int64, selectedVariationIDs: [Int64]) {
+        guard selectedVariationIDs.isNotEmpty else {
+            productIDTrackingSources.removeValue(forKey: productID)
+
+            return
+        }
+
+        productIDTrackingSources[productID] =  retrieveTrackingSource(for: productID)
+    }
```

Comment:
Perhaps we can merge these two into a single method with the `updateTrackingSourceAfterSelectionStateChangedForProduct(with productID: Int64, selectedVariationIDs: [Int64])` signature? We can skip `selectedVariationIDs` logic if the array is empty.

Reply:
After some consideration I preferred to leave it as it is. We ascertain if a variable product is selected by checking on the selectedVariationIDs array; if we also check that for a regular product the logic gets broken. As their logic is therefore different and mostly don't mix, I think it's better to leave it in separate functions to make it simpler.